Tool 3a1

elinevanelburg opened the pull request:
Ik vind dit script behoorlijk ingewikkeld, vandaar dat ik even een review vraag. Ik heb nog niet alle wijzigingen verwerkt, eigenlijk alleen ervoor gezorgd dat de vaste punten naar een eigen shapefile worden uitgeschreven.

bastiaan-roos commented on tools/load_veldwerk_json.py:
```diff
@@ -423,8 +425,52 @@ def fielddata_to_memcollections(filename, profile_plan_col=None, profile_id_fiel
+
+        fixed_point['datum'] = fp.get('datetime', '')
+
+        if fp.get('rd_coordinates') and not None in fp.get('rd_coordinates')[:2]:
```
In de nieuwste versie zit alle informatie over locatie in 'locations' en is er een x y en altitude_nap_pole attribute (z is de hoogte van de gps sensor, hier wordt nog de stok lengte en L1 lengte (afstand tussen stok en gps meter in het apparaat) afgehaald).
ik zou daarom een regel hierboven schrijven waar de x, y, altitude_nap_pole bepaald op basis van de 'location' als die aanwezig is of op basis van de rd_coordinaten[0], rd_coordinaten[1] en altitude_nap_pole.

elinevanelburg replied:
Dit geldt alleen voor de point_notes of ook voor de rest van de json?

bastiaan-roos replied:
dit geldt ook voor de profielen

elinevanelburg replied:
Is hier al een voorbeeld van? Ik vind het een beetje lastig om het zo voor me te zien in het geval van de profielen

elinevanelburg replied:
Heb de code aangepast aan de nieuwe versie

bastiaan-roos commented on tools/load_veldwerk_json.py:
```diff
@@ -423,8 +425,52 @@ def fielddata_to_memcollections(filename, profile_plan_col=None, profile_id_fiel
+            else:
+                fixed_point['z'] = -9999
+            coordinates = (rd_coordinates[0], rd_coordinates[1])
+
```
Wat gebeurt er met point_notes zonder coordinaten?

elinevanelburg replied:
Dat stukje was ik vergeten te programmeren inderdaad. Na overleg met Iris besloten ze wel te laten zien in de shapefile, met de coordinaten (0,0), waardoor het punt dus ergens in Frankrijk terecht komt. De attributen van de coordinaten in de tabel komen te staan als -9999 voor zowel x, y als z. Ik vind het niet een hele mooie oplossing eerlijk gezegd.
Iris zegt eigenlijk dat het gewoon überhaupt niet moet mogen voorkomen.

bastiaan-roos replied:
is inderdaad niet mooi. QGIS kan bijvoorbeeld omgaan met None waarden, maar arcGIS niet.
z = null kan voorkomen als ze een punt op de kaart kiezen x en y kunnen null zijn als een punt wordt toegevoegd, terwijl er nog geen gps punt is ontvangen.

bastiaan-roos commented on tools/load_veldwerk_json.py:
```diff
@@ -426,7 +426,7 @@ def fielddata_to_memcollections(filename, profile_plan_col=None, profile_id_fiel
 
     # Extract fixed points from the geojson
-    for fp_pk, point_ids in enumerate(json_dict['point_notes']):
+    for fp_pk, point_ids in enumerate(json_dict.get('point_notes', [])):
```
Ik zag net in de testdata dat het een dictionary is. 
Volgens mij moet het dan zijn:
json_dict.get('point_notes', {}).items()  of werkt het zo ook?

elinevanelburg replied:
Ik heb het veranderd en wat extra tests toegevoegd om zowel een lege als een niet bestaande point_notes te testen. Ik vermoed dat het daarvoor ook gewoon werkte, ik ondervond geen problemen. Dit is wel mooier natuurlijk.

elinevanelburg commented:
Nu kunnen ook vaste punten worden omgezet met de nieuwe app. De tests falen nog, doordat de code natuurlijk sterk veranderd is. Daar ga ik nu aan werken, maar aangezien dit een beetje snel moet, commit ik maar even falende code.

Verder is er misschien iets raars aan de hand met de vaste punten (de point_notes). Het eerste punt van de serie heeft de opmaak van de vaste punten zoals ik die gewend was bij de laatste versie. De punten erna echter hebben de opmaakt van profielpunten. Hieronder een voorbeeld van Scheldestromen, van de eerste twee point notes. De punten erna zien er allemaal uit als het tweede punt in het voorbeeld.

"point_notes": {
		"0": {
			"note_type": "",
			"remarks": "IJk 549B (1.617)",
			"level": 1.5941435702626663,
			"location": {
				"latitude": 51.301226944833324,
				"longitude": 3.6618197013333336,
				"accuracy": "0.005",
				"altitude": 48.153,
				"altitudeAccuracy": "0.012",
				"metadata": {
					"pole_length": 2,
					"l_one_length": "0.088"
				},
				"x": 34679.90551249965,
				"y": 369417.00712128193,
				"z": 3.6821435702626664,
				"epsg": 28992,
				"datetime": "2018-01-29T10:27:00.569Z",
				"altitude_nap_pole": 1.5941435702626663
			},
			"photos": [
				"file:///storage/emulated/0/Android/data/com.roosgeo.tapp/cache/1517221607500.jpg"
			],
			"created": "2018-01-29T10:27:02.356Z",
			"id": "0"
		},
		"1": {
			"created": "2018-01-29T10:27:30.064Z",
			"last_modified": "",
			"code": "",
			"sub_code": "",
			"drawing_code": "",
			"distance": "",
			"distance_source": "",
			"distance_accuracy": "",
			"distance_manual": "",
			"lower_level": "",
			"lower_level_source": "",
			"lower_level_accuracy": "",
			"lower_level_unit": "",
			"lower_level_manual": "",
			"upper_level": "",
			"upper_level_source": "",
			"upper_level_accuracy": "",
			"upper_level_unit": "",
			"upper_level_manual": "",
			"remarks": "IJk 549B (1.617), 29-01-18",
			"metadata": {
				"method": ""
			},
			"location": {
				"latitude": 51.301226849166675,
				"longitude": 3.6618202739999997,
				"accuracy": "0.005",
				"altitude": 48.156,
				"altitudeAccuracy": "0.012",
				"metadata": {
					"pole_length": 2,
					"l_one_length": "0.088"
				},
				"x": 34679.945191087616,
				"y": 369416.99553608085,
				"z": 3.6851434584155944,
				"epsg": 28992,
				"datetime": "2018-01-29T10:27:28.598Z",
				"altitude_nap_pole": 1.5971434584155944
			},
			"photos": [],
			"note_type": "",
			"level": 1.5971434584155944,
			"id": "1"
		},

Ik kan de data er gewoon uithalen zonder dat de code ervoor hoeft aangepast te worden, dus dat is mooi, maar het is natuurlijk wel een beetje raar.

bastiaan-roos commented:
De eerste point_notes zijn nog gemaakt met een oude versie, de latere met de nieuwste versie (die heb ik inderdaad consistent gemaakt met point_notes).

elinevanelburg commented:
Tests are now fixed, so branch no longer has errors. I changed a lot of other things in the meantime, I am not sure if everything has to be reviewed. However, if the code is okay (okay enough anyway) now, it can be merged. There is also still a pull request waiting in TijhuisArcGisTools.
